src/thorlabs_cube/driver/tcube/tsc.py

```diff
@@ -0,0 +1,360 @@
+
+        :param msg_id: The message ID of the message containing the parameters
+                       that need to be saved in the EEPROM.
+        """
+
+        payload = st.pack("<HH", _CHANNEL, msg_id)
+
+        await self.send(Message(MGMSG.MOT_SET_EEPROMPARAMS, data=payload))
+
+    async def get_status_update(self):
+        """Returns a status update on the specified motor channel
+
+        This function sends a request to the device to return status updates of
+        position, encoder count, status bits on channel one as well as, for future use,
+        returns channel identity two and its associated data which is garbage right now
+
+        :return: integers defining the status bits
+
+        """
+
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_STATUSUPDATE, [MGMSG.MOT_GET_STATUSUPDATE], _CHANNEL
+        )
+
+        (
+            position,
+            encoder_count,
+            status_bits,
+            chan_identity_two,
+            _RESERVED,
+            _RESERVED,
+            _RESERVED,
+        ) = st.unpack("<HIIIHIII", get_msg.data[2:])
+
+        return position, encoder_count, status_bits, chan_identity_two
+
+    async def set_sol_operating_mode(self, operating_mode):
+        """Set the solenoid operating mode for the single channel.
+
+        :param operating_mode: The operating mode to set (e.g., 1, 2, etc.).
+        """
+
+        await self.send(Message(MGMSG.MOT_SET_SOL_OPERATINGMODE, param2=operating_mode))
+
+    async def get_sol_operating_mode(self):
+        """Get the current solenoid operating mode for the single channel.
+
+        :return: operating mode of solenoid represented by an integer
+        """
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_OPERATINGMODE,
+            [MGMSG.MOT_GET_SOL_OPERATINGMODE],
+            param1=_CHANNEL,
+        )
+
+        return get_msg.param2
+
+    async def set_solenoid_cycle_parameters(self, on_time, off_time, num_cycles):
+        """Set the solenoid cycle parameters.
+
+        :param on_time: Time (in ms) the solenoid stays on (100ms to 10,000ms).
+        :param off_time: Time (in ms) the solenoid stays off (100ms to 10,000ms).
+        :param num_cycles: Number of open/close cycles (0 for infinite, up to 1,000,000).
+        """
+
+        payload = st.pack("<HLLL", _CHANNEL, on_time, off_time, num_cycles)
+
+        await self.send(Message(MGMSG.MOT_SET_SOL_CYCLEPARAMS, data=payload))
+
+    async def get_solenoid_cycle_parameters(self):
+        """Get the current solenoid cycle parameters.
+
+        :return: A tuple containing (on_time, off_time, num_cycles).
+        """
+
+        payload = st.pack("<H", _CHANNEL)
+
+        get_msg = await self.send_request(
+            MGMSG.MOT_REQ_SOL_CYCLEPARAMS, [MGMSG.MOT_GET_SOL_CYCLEPARAMS], data=payload
+        )
+
+        on_time, off_time, num_cycles = st.unpack("<LLL", get_msg.data[2:])
+
+        return on_time, off_time, num_cycles
```

Comment:
```suggestion
        return st.unpack("<LLL", get_msg.data[2:])
```
Unpacking the values then returning them unmodified is unnecessary.